src/resolve-dependency.js

```diff
@@ -11,7 +11,22 @@ module.exports = function resolveDependency (specifier, parent, job) {
   else
     resolved = resolvePackage(specifier, parent, job);
   if (resolved.startsWith('node:')) return resolved;
-  return fs.realpathSync(resolved);
+
+  const real = fs.realpathSync(resolved);
+  const original = resolved;
+  let isLink = false;
+  
+  if (resolved !== real) {
+    while (resolved !== '/') {
+      // walk up the tree to confirm that this is a symlink
```

Comment:
I think we can save some `lstat` executions here and just compare `resolve` and `real` to extract their "common part", and look for a symlink there.

Reply:
We can probably assume any difference here is a symlink and emit so I pushed a change.